src/com/dotmarketing/portlets/rules/business/RulesEngine.java

```diff
@@ -44,13 +50,32 @@ public static void fireRules(HttpServletRequest req, HttpServletResponse res, Ru
 
         try {
 
-            Set<Rule> rules = APILocator.getRulesAPI().getRulesByParentFireOn(host.getIdentifier(), systemUser, false, fireOn);
-
+            Set<Rule> rules = APILocator.getRulesAPI().getRulesByParentFireOn(parent.getIdentifier(), systemUser, false, fireOn);
+			List<Rule> firedRules = (List<Rule>) req.getAttribute(WebKeys.RULES_ENGINE_FIRE_LIST);
+        	if(firedRules ==null){
+        		firedRules = new ArrayList<Rule>();
+        		req.setAttribute(WebKeys.RULES_ENGINE_FIRE_LIST, firedRules);
+        	}
```

Comment:
Need to keep in mind that not all rules fire on a per request/page basis - believe that some rules firing status will need to be kept in the session - i.e. for "Once per visit" and "Once per visitor" firing frequency will cause rules to fire that may affect all of the requests in this session (by altering state) but will only fire on the first request of the users session